### PR TITLE
feat: fix password registry leak of credentials

### PR DESCRIPTION
**Pull request by zhaque44**
#1669

### CODE REVIEW

File: pkg/helmexec/exec.go [hunk 1]
```diff
@@ -186,22 +188,30 @@ func (helm *execer) AddRepo(name, repository, cafile, certfile, keyfile, usernam
...
+		if username != "" && password != "" {
+			args = append(args, "--username", username, "--password-stdin")
+			buffer := bytes.Buffer{}
+			buffer.Write([]byte(fmt.Sprintf("%s\n", password)))
+			out, err = helm.execStdIn(args, map[string]string{}, &buffer)
+		} else {
+			out, err = helm.exec(args, map[string]string{}, nil)
+		}
+
 		helm.logger.Infof("Adding repo %v %v", name, repository)
```
Comment (yxxhero) [inline pkg/helmexec/exec.go:208]: move this code to line 198.

Comment (yxxhero) [inline pkg/helmexec/exec.go:208]:
```suggestion
		helm.logger.Infof("Adding repo %v %v", name, repository)
		if username != "" && password != "" {
			args = append(args, "--username", username, "--password-stdin")
			buffer := bytes.Buffer{}
			buffer.Write([]byte(fmt.Sprintf("%s\n", password)))
			out, err = helm.execStdIn(args, map[string]string{}, &buffer)
		} else {
			out, err = helm.exec(args, map[string]string{}, nil)
		}
```

**Comment by yxxhero** [comment]
```
== RUN   Test_AddRepo
    exec_test.go:232: helmexec.AddRepo()
        actual = Adding repo myRepo https://repo.example.com/
        exec: helm --kubeconfig config --kube-context dev repo add myRepo https://repo.example.com/ --username example_user --password-stdin
        
        expect = Adding repo myRepo https://repo.example.com/
        exec: helm --kubeconfig config --kube-context dev repo add myRepo https://repo.example.com/ --username example_user --password example_password
    exec_test.go:256: helmexec.AddRepo()
        actual = Adding repo myRepo https://repo.example.com/
        exec: helm --kubeconfig config --kube-context dev repo add myRepo https://repo.example.com/ --pass-credentials --username example_user --password-stdin
        
        expect = Adding repo myRepo https://repo.example.com/
        exec: helm --kubeconfig config --kube-context dev repo add myRepo https://repo.example.com/ --username example_user --password example_password --pass-credentials
```

**Comment by yxxhero** [comment]
@zhaque44 please fix failed test case. just only update the `expect code`.

**Comment by yxxhero** [comment]
```
=== RUN   Test_LogLevels
    exec_test.go:932: helmexec.AddRepo()
        actual = Adding repo myRepo https://repo.example.com/
        exec: helm repo add myRepo https://repo.example.com/ --username example_user --password-stdin
        
        expect = Adding repo myRepo https://repo.example.com/
        exec: helm repo add myRepo https://repo.example.com/ --username example_user --password example_password
```

**Comment by zhaque44** [comment]
fixed @yxxhero we have a minor DCO commit issue: https://github.com/helmfile/helmfile/pull/1687/commits/bef177110f1df9a122caacc35a245e6f04655b69 <- you did not sign this commit. Everything else passed

**Review by yxxhero — APPROVED** [review]

**Pull request merged**